### DISCUSSION TITLE
DAI 2024-2025 - Java IOs benchmarking - Yoann Changanaqui

### DISCUSSION THREAD

**Discussion by Yoy017** [heig-vd-dai-course/heig-vd-dai-course#372] [Show and tell]
[Link to the repo](https://github.com/Yoy017/05-java-ios)
### Write

| Size of file | Binary | Buf binary | Text | Buf text |
| ------------ | ------ | ---------- | ---- | -------- |
| **1B**       | 144    | 140        | 128  | 128      |
| **1KiB**     | 160    | 140        | 134  | 133      |
| **1MiB**     | 1191   | 232        | 262  | 163      |
| **5MiB**     | 5191   | 252        | 345  | 250      |

*(Time in ms)*

### Read

| Size of file | Binary | Buf binary | Text | Buf text |
| ------------ | ------ | ---------- | ---- | -------- |
| **1B**       | 127    | 124        | 131  | 129      |
| **1KiB**     | 290    | 127        | 227  | 132      |
| **1MiB**     | 435    | 201        | 260  | 161      |
| **5MiB**     | 1508   | 1496       | 427  | 248      |
*(Time in ms)*

#### Which type of stream is the most efficient for each use case?
The most efficient type is the `Buffered`.

#### Why is it more efficient than the other types of streams?
Because the buffer make the program use way less call of `read()` or `write()`. It improve the performance since the call of these method are few compare to the other types of streams.

#### What is the difference between binary data and text data?
A `Text data` file uses an character encoding that map binary code to a character. Meanwhile `binary data` file contains raw binary. It is not encoded and non-human readable. In general, binary data file are compiled programs.

#### What is a character encoding?
It is use to assign a binary code to a specific symbol. It's use for character and emojis. One of the most well-known is ASCII.

#### Why is this benchmark methodology important?
First to experiment writing IOs files that will be usefull for the `Practical work 1` of this course. Secondly to see in what way one specific method *(buffered)* is the most efficient !

**Comment by ludelafo** [comment]
Thank you!

You can always find the solution in the solution repository if you ever need to compare your work with the official one here: https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions.

Do not hesitate to ask questions if something is not clear, either in this discussion or in the Q&A category of the GitHub Discussions here: https://github.com/orgs/heig-vd-dai-course/discussions/categories/q-a.
